File: Two_pointers/Find_sub-array_with_the_given_sum/Find_sub-array_with_the_given_sum.java
public class Solution {
	public int[] solve(int[] A, int B) {
		int size = A.length;
		long sum = A[0];
		int l = 0,
		r = 1;
		while (l <= r) {
			if (sum == B) {
				int[] found = new int[r - l];
				for (int x = l; x < r; x++) {
					found[x - l] = A[x];
				}
				return found;
			}
			else if (sum < B && r != size) {
				sum += A[r++];
			}
			else {
				if (l == size) {
					break;
				}
				sum -= A[l++];
			}
		}
		int[] notFound = { - 1
		};
		return notFound;
	}
}

File: Two_pointers/Find_sub-array_with_the_given_sum/Find_sub-array_with_the_given_sum.cpp
vector<int> Solution::solve(vector<int> &A, int B)
{
    int size = A.size();
    long sum = A[0];
    int l = 0, r = 1;
    while (l <= r)
    {
        if (sum == B)
        {
            vector<int> ans;
            for (int i = l; i < r; i++)
            {
                ans.push_back(A[i]);
            }
            return ans;
            // l  :   0-> 1
            // sum :  6-1 = 5
        }
        else if (sum < B && r != size)
        {
            sum += A[r++];
            //  r :    1 2->3
            //  sum :  3 6
        }
        else
        {
            sum -= A[l++];
        }
        //  Case 1 : When Sum equals B;
        //  Case 2 : When Sum Less Than B ;
        //  Case 3 : When Sum Greater Than B;
        //  Case 4 : When Sum Equals The First Number Of the Array .
        // [10, 0]
        //  Case 5 : When Sum Ends At The Last Character of the Array .
        // [1 0 , 0 , 10 ] = [0 ,0, 10]
    }
    return {-1};
}
